docs/src/main/asciidoc/hibernate-reactive.adoc

```diff
@@ -0,0 +1,229 @@
+
+[source,properties]
+.Example `{config-file}`
+----
+# datasource configuration
+quarkus.datasource.db-kind = postgresql
+quarkus.datasource.username = quarkus_test
+quarkus.datasource.password = quarkus_test
+
+quarkus.datasource.reactive.url = vertx-reactive:postgresql://localhost/quarkus_test <1>
+
+# drop and create the database at startup (use `update` to only update the schema)
+quarkus.hibernate-orm.database.generation=drop-and-create
+----
+
+<1> The only different property from a Hibernate ORM configuration
+
+Note that these configuration properties are not the same ones as in your typical Hibernate Reactive configuration file.
+They will often map to Hibernate Reactive configuration properties but could have different names and don't necessarily map 1:1 to each other.
+
+Also, Quarkus will set many Hibernate Reactive configuration settings automatically, and will often use more modern defaults.
+
+WARNING: Configuring Hibernate Reactive using the standard `persistence.xml` configuration file is not supported.
+
+Please, see section <<hr-configuration-properties, Hibernate Reactive configuration properties>> for the list of properties you can set in `{config-file}`.
+
+A `Mutinty.SessionFactory` will be created based on the Quarkus `datasource` configuration as long as the Hibernate Reactive extension is listed among your project dependencies.
+
+The dialect will be selected based on the Vert.x SQL client - unless you set one explicitly.
+
+You can then happily inject your `Mutiny.SessionFactory`:
+
+[source,java]
+.Example application bean using Hibernate Reactive
+----
+@ApplicationScoped
+public class SantaClausService {
+    @Inject
+    Mutiny.SessionFactory sf; <1>
+
+    public Uni<Void> createGift(String giftDescription) {
+	Gift gift = new Gift();
+        gift.setName(giftDescription);
+	return sf.withTransaction(session -> session.persist(gift)) <2>
+    }
+}
+----
+
+<1> Inject your session factory and have fun
+<2> `.withTransaction` will automatically flush at commit
+
+[source,java]
+.Example of an Entity
+----
+@Entity
+public class Gift {
+    private Long id;
+    private String name;
+
+    @Id
+    @SequenceGenerator(name = "giftSeq", sequenceName = "gift_id_seq", allocationSize = 1, initialValue = 1)
+    @GeneratedValue(generator = "giftSeq")
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+}
+----
+
+To load SQL statements when Hibernate Reactive starts, add an `import.sql` file to the root of your resources directory.
+This script can contain any SQL DML statements.
+Make sure to terminate each statement with a semicolon.
+
+This is useful to have a data set ready for your tests or demos.
+
+WARNING: Make sure to wrap methods modifying your database (e.g. `session.persist(entity)`) within a transaction. 
```

Comment:
I think this warning is a bit misplaced? It has nothing to do with the `import.sql`.